packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -201,30 +201,23 @@ export class SearchBoxBase extends React.Component<ISearchBoxProps, ISearchBoxSt
     switch (ev.which) {
       case KeyCodes.escape:
         this.props.onEscape && this.props.onEscape(ev);
-        if (!ev.defaultPrevented) {
+        if (this.state.value && !ev.defaultPrevented) {
           this._onClear(ev);
         }
         break;
 
       case KeyCodes.enter:
         if (this.props.onSearch) {
           this.props.onSearch(this.state.value);
-          break;
+          ev.preventDefault();
+          ev.stopPropagation();
         }
-        // if we don't handle the enter press then we shouldn't prevent default
-        return;
+        break;
 
       default:
         this.props.onKeyDown && this.props.onKeyDown(ev);
-        if (!ev.defaultPrevented) {
-          return;
-        }
+        break;
```

Comment:
To mirror the previous logic, I think you'd need to do `if (ev.defaultPrevented) { ev.stopPropagation(); }` here? Unless that had undesirable effects or you think it was unnecessary to call previously.

Reply:
It does seem weird and unexpected that we'd call stopPropagation even if the custom onKeyDown handler did not. But given that there's no known issue with doing it, I'll add back the stopPropagation call to avoid risking a regression if anything was relying on that behavior.

Reply:
Same thought here...I agree it's weird but don't want to break stuff.